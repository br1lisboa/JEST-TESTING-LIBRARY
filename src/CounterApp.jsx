import { useState } from 'react';
import PropTypes from 'prop-types';

export default function CounterApp({ value }) {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => {
        //setCounter(counter + 1)
        setCounter((c) => c + 1)
    }
    const handleRest = () => {
        setCounter(value)
        //setCounter((c) => c = 0)
    }
    const handleSub = () => {
        //setCounter(counter + 1)
        setCounter((c) => c - 1)
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <div className='btn-container'>
                <button onClick={handleAdd}>+1</button>
                <button aria-label='btn-reset' onClick={handleRest}>Reset</button>
                <button onClick={handleSub}>-1</button>
            </div>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}