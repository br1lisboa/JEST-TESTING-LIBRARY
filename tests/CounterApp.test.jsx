import { fireEvent, render, screen } from "@testing-library/react"
import CounterApp from "../src/CounterApp"


describe('Probando CounterApp ', () => {

    const value = 1

    test('CounterApp debe hacer match con el snapshot', () => {

        const { container } = render(<CounterApp value={value} />)
        //console.log(container);
        expect(container).toMatchSnapshot()

    })

    test('Debe de mostrar el valor inicial de 100', () => {

        render(<CounterApp value={100} />)

        expect(screen.getByText(100)).toBeTruthy

    })

    test('Debe de incrementar con el boton +1', () => {

        render(<CounterApp value={value} />)

        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('2')).toBeTruthy()

    })

    test('Debe de decrementar con el boton de -1', () => {

        render(<CounterApp value={value} />)

        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('0')).toBeTruthy()

    })

    test('Debe de funcionar el boton de RESET', () => {

        render(<CounterApp value={value} />)

        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        // fireEvent.click(screen.getByText('Reset'))
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))
        expect(screen.getByText(1)).toBeTruthy()

    })

})