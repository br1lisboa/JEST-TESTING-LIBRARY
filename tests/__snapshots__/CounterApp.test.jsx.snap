// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando CounterApp  CounterApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    1
  </h2>
  <div
    class="btn-container"
  >
    <button>
      +1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
    <button>
      -1
    </button>
  </div>
</div>
`;
